content/blog/2017-11-28-fragments-in-react-v16.2.md

```diff
@@ -0,0 +1,181 @@
+
+```jsx
+render() {
+  return (
+    <>
+      <span>I am</span>
+      <span>static</span>
+    </>
+  );
+}
+```
+
+#### The Syntax
+
+Inspired by some prior art such as the `XMLList() <></>` constructor in [E4X](https://developer.mozilla.org/en-US/docs/Archive/Web/E4X/E4X_for_templating) and the existing `<></>` syntax in [Reason React](https://reasonml.github.io/reason-react/), the React team concluded on having a convenient syntax for JSX fragments that looks like `<>...</>`. This captures the idea of a fragment: It is a pair of empty tags which enforces the notion that it won't render to an actual element in the DOM.
+
+The `<></>` syntax also makes it easier to refactor since it looks just like a JSX element. For example, consider swapping the parents in the following code snippets:
+
+```jsx
+// with a parent wrapper
+<div>
+  <span>foo</span>
+  bar
+  <span>baz</span>
+</div>
+
+// with arrays (where commas are necessary and text needs to be surrounded by quotes)
+[
+  <span>foo</span>,
+  'bar',
+  <span>baz</span>
+]
+
+// with jsx fragment syntax
+<>
+  <span>foo</span>
+  bar
+  <span>baz</span>
+</>
+```
+
+#### The Export
+
+After the release of React 16, people in the React community have created solutions in order to address the problem of static lists, such as [react-aux](https://github.com/gajus/react-aux). For `Fragment`s however, the React team has decided to take ownership of the export, since there are some semantics specific to Fragments that aren't possible without core library changes. These changes have some performance wins and make reconciliation a bit more intuitive.
+
+[For the curious, you can read more about that here.](https://github.com/facebook/react/pull/10783)
+
+Still, we are very thankful to the open source contributors who worked on solving this in the ecosystem.
+
+Additionally, there is a [pragma option available in babel-plugin-transform-react-jsx](https://github.com/babel/babel/tree/master/packages/babel-plugin-transform-react-jsx#pragmafrag) that allows the Babel compiler to de-sugar the `<>...</>` syntax to a custom identifier.
+
+### Example Usages
+
+As mentioned above, the most common way to use fragments in React is to simply use `<></>` tags. Generally, in the Javascript compiler phase of your build pipeline, the `<>...</>` will be de-sugared into a `createElement` call with `React.Fragment` as the type.
+
+Take a look at the section on [How to Use it](#how-to-use-it) to see how to update your tooling to support JSX fragments.
+
+#### With Attributes and Keys
+
+Note that with the `<></>` syntax, attributes are not allowed to be passed. We made the decision to disallow attributes with fragments since they're semantically different from React elements. However, a fragment can be keyed by using the explicit fragment syntax:
+
+```jsx
+import { Fragment } from 'react';
```

Comment:
Can we remove this and just use `React.Fragment`? Thinking about ES6 imports here adds unnecessary overhead to an already new concept.

Reply:
@sebmarkbage 's thought is that `React.Fragment` is quite long and will be harder to sell.